Automattic/wp-calypso
# Posts: Convert PostSelector to using posts Redux state

Related: #303, #2248

__Edit:__ The purpose of this pull request has changed significantly. Refer to [this gist](https://gist.github.com/aduth/a7e22040dff95702dbb0) for the original pull request content.

This pull request seeks to convert the existing `<PostSelector />` component to make use of the global Redux post state. It introduces a new `<QueryPosts />` component for managing requests to the REST API, and implements the prerequisite pagination behavior to the posts state.

__Implementation notes:__

An advantage of the new posts state is that previous post queries are cached, so if a user enters a search term, then clears their query, the component no longer needs to make another request for the original posts.

The `<QueryPosts />` component is the first example of a [proposed fetching components pattern](https://github.com/Automattic/wp-calypso/blob/add/docs-data-approach/docs/our-approach-to-data.md#fetching-components) in managing the fetching of data to be accessed via global application state by sibling components.

__Testing instructions:__

Confirm that there are no regressions with the parent page selector in the Calypso post editor, including infinite scroll pagination, search queries, and lack of results/pages.

1. Navigate to the [Calypso page editor](http://calypso.localhost:3000/page)
2. Select a site, if prompted
3. Expand the Page Options sidebar accordion
4. Note that available pages are shown, ordered ascending by title, excluding the current page (if published)
5. Note that you can scroll the parent page selector for infinite scroll if many pages exist (more than 20)
6. Note that entering a search term limits the shown posts to the filtered search term

Also confirm that Mocha tests pass by running `make test` from the project root directory or directly from `client/state/posts`.
